workerapi/api.go

```diff
@@ -1,4 +1,89 @@
 package workerapi
 
 // We should use go generate to run:
-//   thrift --gen go:package_prefix=github.com/scootdev/scoot/scootapi/gen-go/,thrift_import=github.com/apache/thrift/lib/go/thrift worker.thrift
+//   thrift --gen go:package_prefix=github.com/scootdev/scoot/workerapi/gen-go/,thrift_import=github.com/apache/thrift/lib/go/thrift worker.thrift
+
+import (
+	"time"
+
+	"github.com/scootdev/scoot/runner"
+	"github.com/scootdev/scoot/workerapi/gen-go/worker"
+)
+
+//
+// Translation between local domain objects and thrift objects:
+//
+
+func ThriftRunCommandToDomain(thrift *worker.RunCommand) *runner.Command {
+	var timeout time.Duration
+	if thrift.TimeoutMs != nil {
+		timeout = time.Millisecond * time.Duration(*thrift.TimeoutMs)
+	}
+	return runner.NewCommand(thrift.Argv, thrift.Env, timeout)
+}
+
+func DomainRunCommandToThrift(domain *runner.Command) *worker.RunCommand {
+	thrift := worker.NewRunCommand()
+	timeoutMs := int32(domain.Timeout / time.Millisecond)
+	thrift.TimeoutMs = &timeoutMs
+	thrift.Env = domain.EnvVars
+	thrift.Argv = domain.Argv
+	return thrift
+}
+
+func ThriftRunStatusToDomain(thrift *worker.RunStatus) *runner.ProcessStatus {
+	domain := &runner.ProcessStatus{}
+	domain.RunId = runner.RunId(thrift.RunId)
+	switch thrift.Status {
+	case worker.Status_UNKNOWN:
+		domain.State = runner.UNKNOWN
+	case worker.Status_PENDING:
+		domain.State = runner.PENDING
+	case worker.Status_RUNNING:
+		domain.State = runner.RUNNING
+	case worker.Status_FAILED:
+		domain.State = runner.FAILED
+	case worker.Status_ABORTED:
+		domain.State = runner.ABORTED
+	case worker.Status_TIMEDOUT:
+		domain.State = runner.TIMEDOUT
+	}
+	if thrift.OutUri != nil {
+		domain.StdoutRef = *thrift.OutUri
+	}
+	if thrift.ErrUri != nil {
+		domain.StderrRef = *thrift.ErrUri
+	}
+	if thrift.Error != nil {
+		domain.Error = *thrift.Error
+	}
+	if thrift.ExitCode != nil {
+		domain.ExitCode = int(*thrift.ExitCode)
+	}
+	return domain
+}
+
+func DomainRunStatusToThrift(domain *runner.ProcessStatus) *worker.RunStatus {
```

Comment:
Yea whatever is easiest here for testing is nice since the statespace is fairly limited a table test can work just as well!